Packages/com.unity.inputsystem/InputSystem/InputManager.cs

```diff
@@ -3808,9 +3829,11 @@ internal void RestoreStateWithoutDevices(SerializedState state)
                 Object.DestroyImmediate(m_Settings);
             m_Settings = state.settings;
 
-            if (m_Actions != null)
-                Object.DestroyImmediate(m_Actions);
+            #if UNITY_INPUT_SYSTEM_PROJECT_WIDE_ACTIONS
+            // Note that we just reassign actions and never destroy them since always mapped to persisted asset
```

Comment:
This didn't make any sense to keep since when triggered it would attempt to destroy an asset object associated with an asset in edit mode  (DestroyImmediate), hence why I removed it. It got triggered in unit testing which made ADB scream at me that you should never do this and the public docs says the same. 
It makes a lot of since Unity owns and manages the instance. Only programatically created objects should be deleted like this.
Basically: Since we only allow persisted assets to be assigned we should never destroy the object is at least my interpretation.